api/client/network.go

```diff
@@ -206,12 +217,28 @@ func (cli *DockerCli) CmdNetworkInspect(args ...string) error {
 			status = 1
 			continue
 		}
-		networkResource := types.NetworkResource{}
+		var networkResource interface{}
 		if err := json.NewDecoder(bytes.NewReader(obj)).Decode(&networkResource); err != nil {
 			return err
 		}
 
-		networks = append(networks, &networkResource)
+		if tmpl == nil {
+			networks = append(networks, &networkResource)
+			continue
+		}
+
+		if err := tmpl.Execute(cli.out, &networkResource); err != nil {
+			if err := tmpl.Execute(cli.out, &networkResource); err != nil {
```

Comment:
you shouldn't deserialize directly on `cli.out`, otherwise you might end up printing partial results.